### DISCUSSION TITLE
how to add more layout and set time to 24 format by default

### DISCUSSION THREAD

**Discussion by ghost** [prasanthrangan/hyprdots#203] [Q&A]
how to add more layout and set time to 24 format by default

**Comment by T-Crypt** [comment]
@crawbrokes39 explain add more layout? What do you mean by add more layout??

Just left click on the time displayed on the waybar to convert to 24 hr format.

**Comment by ghost** [comment]
@T-Crypt i need to add russian and ukranian layout and i want 24 hr time format by default on bar

**Reply by T-Crypt** [reply]
1. You need to modify the clock module for waybar on your local machine 

location: $HOME/(user)/.config/waybar/modules/clock.jsonc

     "clock": {
        "format": "{: %I:%M %p 󰃭 %a %d}", >>>>> add to "format-alt"
        "format-alt": "{:󰥔 %H:%M  %b %Y}", >>>>> add to "format"
        "tooltip-format": "<tt><big>{calendar}</big></tt>"
    },


2.  you need to change keyboard layout in hyprland.conf 

location: $HOME/(user)/.config/hypr/hyprland.conf

input {
    kb_layout = us (change "us" to desired layout)
    kb_variant =
    kb_model =
    kb_options =
    kb_rules =
    follow_mouse = 1

**Comment by ghost** [comment]
@T-Crypt but i need to add more than 1 layout i cant do this in hyprland config maybe i do something wrong?

**Reply by T-Crypt** [reply]
You can.

input {
    kb_layout = ru,ua
    kb_options = grp:alt_shift_toggle
}

Then the whichever layout is declared first is the layout for the keybindings (so for this it would go off the ru layout)

https://wiki.hyprland.org/Configuring/Uncommon-tips--tricks/

**Reply by T-Crypt** [reply]
You can also bind a key to execute hyprctl switchxkblayout for more keybind freedom.

(https://wiki.hyprland.org/Configuring/Using-hyprctl).

**Reply by ghost** [reply]
@T-Crypt with time format i did everything as you said but nothing changed, and I would also like to know if it is possible to make a notification about the change of layout and on waybar that it would be visible what is the current layout.

**Reply by T-Crypt** [reply]
1. For the clock: You need to run the wbarconfgen.sh to create the new conf file for waybar.

Just use the keybindings SUPER + ALT + UP ARROW or DOWN ARROW to redo the waybar config with the changes in the clock.jsonc module. Then the format will be changed to 24 hour as default and 12 hr as alt.

2. Ill have to see if there is a way to notify / show the current KB layout. You may need to give me a second for this!

**Reply by T-Crypt** [reply]
You can use 

`hyprctl devices` to get the output of your connected devices and view the current KB layout.

Can also use `hyprctl switchxkblayout [DEVICE] [CMD]`

^^^^^^
(where [DEVICE] is from the hyprctl devices and [CMD] = 'next' or 'prev' or using the id of the country code 'ru' or 'au')

EXAMPLE: `hyprctl switchxkblayout asus-keyboard next`

**Reply by ghost** [reply]
@T-Crypt sorry i'm not very experienced in this topic I'm just starting to use window managers can you be more specific about what i need to do?

**Reply by T-Crypt** [reply]
Did you change your keyboard layout in the hyprland.conf? 

You can use visual studio code to edit the file or use `sudo nano hyprland.conf` in the terminal. 

location: $HOME/(user)/.config/hypr/hyprland.conf 

(you will need to have view hidden files enabled to find .config in the file manager / dolphin)

**Comment by ghost** [comment]
@T-Crypt 
i have added new layouts i would like just a notification about the change of layout as for example about the change of volume just difficult to navigate with 3 layouts

**Reply by T-Crypt** [reply]
@crawbrokes39 going to have to add to the dunst config to notify on kb layout change. 

@prasanthrangan im not as familiar with dunst as you, able to assist?

**Reply by prasanthrangan** [reply]
Hi, can you try updating `~/.config/waybar/modules/clock.jsonc` as -

```
 "clock": {
    "format": "{: %I:%M %p 󰃭 %a %d}",
    "format-alt": "{:󰥔 %H:%M  %b %Y}",
    "tooltip-format": "<tt><big>{calendar}</big></tt>"
},

 "clock#2": {
    "format": "{:󰥔 %H:%M 󰃭 %a %d}",
    "format-alt": "{: %I:%M %p   %b %Y}",
    "tooltip-format": "<tt><big>{calendar}</big></tt>"
},
```
and use clock#2 in `~/.config/waybar/config.ctl` to display it.

**Comment by T-Crypt** [comment]
@crawbrokes39 Hey buddy, the layout change is fixed in #314 

Can now switch between your russian and ukranian layout
